### PR TITLE
Tunneling improvements

### PR DESCRIPTION
**Pull request by LukaszStem**
This PR has the following improvements:

Implemented:
- State clearer message to connect on which port
- Show username and password when server-side ssh port is detected
- Increase ssh timeout from 30 seconds to 30 minutes
- (--port) argument is now optional, if not supplied, will pick a random available port
- ping scm site before creating tunnel
- Have CTRL-C terminate local server

Note: Will update index.json in subsequent commits

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `./scripts/ci/test_static.sh` locally? (`pip install pylint flake8` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli-extensions/blob/master/docs/extension_summary_guidelines.md).

### CODE REVIEW

File: src/webapp/azext_webapp/custom.py [hunk 1]
```diff
@@ -193,6 +193,15 @@ def create_deploy_webapp(cmd, name, location=None, dryrun=False):
     logger.warning("All done.")
     return create_json
 
+def _ping_scm_site(cmd, resource_group, name):
```
Comment (panchagnula) [inline src/webapp/azext_webapp/custom.py]: can you update the create_deploy_webapp to use the helper directly as well.

Comment (LukaszStem) [reply src/webapp/azext_webapp/custom.py]: will do

**Comment by LukaszStem** [comment]
CI is failing because of https://github.com/Azure/azure-cli-extensions/pull/162

**Comment by derekbekoe** [comment]
The other PR has been merged for now

**Pull request merged**